backend/onedrive/onedrive.go

```diff
@@ -1503,7 +1503,78 @@ func (f *Fs) PublicLink(ctx context.Context, remote string, expire fs.Duration,
 		fmt.Println(err)
 		return "", err
 	}
-	return result.Link.WebURL, nil
+
+	url := result.Link.WebURL
+
+	// Convert share link to direct download link if target is not a folder
+	cnvFailMsg := "Don't know how to convert share link to direct link - returning the link as is"
+	if info.Folder != nil {
+		fs.Logf(f, cnvFailMsg)
+		return url, nil
+	}
+
+	segments := strings.Split(url, "/")
+	switch f.driveType {
+	case driveTypePersonal:
+		/*
+			Method: https://stackoverflow.com/questions/37951114/direct-download-link-to-onedrive-file
+		*/
+		if len(segments) != 5 {
+			fs.Logf(f, cnvFailMsg)
+			return url, nil
+		}
+		enc := base64.StdEncoding.EncodeToString([]byte(url))
+		strings.ReplaceAll(enc, "/", "_")
+		strings.ReplaceAll(enc, "+", "-")
+		url = "https://api.onedrive.com/v1.0/shares/u!" + enc[:len(enc)-1] + "/root/content"
```

Comment:
If I understand this correctly, you're just stripping off the last character from the encoded string unconditionally? Does this work when the last character is not a padding character, or if there are two padding characters at the end? (Please feel free to correct me.)

Reply:
Well, actually the Microsoft API works even with the padding character included. But sure, let's strictly follow the format.